File: api/src/main/java/com/superdoodle/api/model/Game.java
package com.superdoodle.api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;


@Data
@AllArgsConstructor
@Builder
public class Game {


    private final int width = 400;

    private final int height = 200;

    private Long Id = 1L;

    private List<Player> players = new ArrayList<>();

    private Fruit fruit;

    public Game(){
        this.fruit = new Fruit(this);
    }


    public void addPlayer(Player player) {
        this.players.add(player);
    }
}

File: app/src/types/Game.ts
import { Fruit, isFruit } from './Fruit';
import { Player, isPlayer } from './Player';

export interface Game {
  id: number;
  players: Player[];
  fruit: Fruit;
  width: number;
  height: number;
}

export function isGame(value: unknown): value is Game {
  return (
    typeof value === 'object' &&
    value !== null &&
    typeof (value as Game).id === 'number' &&
    Array.isArray((value as Game).players) &&
    (value as Game).players.every(isPlayer) &&
    isFruit((value as Game).fruit)
  )
}
